shortcut/base.py

```diff
@@ -15,77 +17,162 @@ class ShortCutter(object):
         s.create_menu_shortcut("python")
     """
 
-    def __init__(self):
+    def __init__(self, err_file=None):
+        self.err_file = sys.stderr if (err_file is None) else err_file
         self._desktop_folder = self._get_desktop_folder()
         self._menu_folder = self._get_menu_folder()
 
     # should be overridden
     def _get_desktop_folder(self):
         raise ShortcutError("_get_desktop_folder needs overriding")
-    
+
     # should be overridden
     def _get_menu_folder(self):
         raise ShortcutError("_get_menu_folder needs overriding")
 
-    def create_desktop_shortcut(self, target):
+    def create_desktop_shortcut(self, target, target_name=None, target_is_dir=False, virtual=True):
         """
         Creates a desktop shortcut to a target.
 
         :param str target:
             The target to create a shortcut for, it can be a fully qualified
             file path `/path/to/my_program` or a simple application name 
             `my_program`.
-
-        Returns a tuple of (target_name, target_path, shortcut_file_path)
+        :param str target_name:
+            Name of the shortcut without extension (.lnk would be appended if needed).
+        :param bool target_is_dir:
+            whether it's a shortcut to a directory
+        :param bool virtual:
+            whether to allow shortcuts to yet non-existing files
```

Comment:
Why should we be making shortcuts to file which dont exist? No problem with the implementation, Im just not sure of the use case.